compiler_opt/es/blackbox_optimizers.py

```diff
@@ -212,44 +226,112 @@ def set_state(self, state: SequenceOfFloats) -> None:
     raise NotImplementedError('Abstract method')
 
 
-class MonteCarloBlackboxOptimizer(BlackboxOptimizer):
-  """Class implementing GD optimizer with MC estimation of the gradient."""
+class CommonStateHandler(BlackboxOptimizer):
```

Comment:
How about `StatefulOptimizer` or `AbstractStatefulOptimizer` - while it _is_ a common something or other, so would all intermediate base classes be - meaning, calling it that doesn't offer any information that's not implicit.

I'm proposing `StatefulOptimizer` because it seems the abstraction brings in state management?

(I realize "state" in CommonStateHandler may actually refer to the algorithm state, but the name also makes it sound so much like "this is a grabbag of some common fields")